index.js

```diff
@@ -1,30 +1,43 @@
 'use strict';
 var got = require('got');
 var registryUrl = require('registry-url');
+var objectAssign = require('object-assign');
 
-module.exports = function (keyword, cb) {
-	got(module.exports.url(keyword), function (err, data) {
+function npmKeywordUrl (keyword, description) {
+	keyword = encodeURIComponent(keyword);
+
+	return registryUrl() +
+		'-/_view/byKeyword?' +
+		'startkey=[%22' + keyword + '%22]' +
+		'&endkey=[%22' + keyword + '%22,%7B%7D]' +
+		'&group_level=' + (description ? 3 : 2);
+}
+
+module.exports = function (keyword, opts, cb) {
+	if (typeof keyword !== 'string') {
+		throw new TypeError('Keyword must be a string');
+	}
+
+	cb = typeof opts === 'function' ? opts : cb;
+
+	opts = objectAssign({
+		description: true
+	}, opts);
+
+	got(npmKeywordUrl(keyword, opts.description), function (err, data) {
 		if (err) {
 			cb(err);
 			return;
 		}
 
-		var rows = JSON.parse(data).rows;
```

Comment:
extra var wasn't necessary since 14-16 werent necessary as the map would already return an empty array when the rows were empty. probably should be a check to make sure `JSON.parse(data).rows ` returns and returns an array though.